### ISSUE TITLE
Test suite failed to run due to: SyntaxError: Unexpected token {

### ISSUE THREAD

**Issue by StefanoA1** [eddyerburgh/vue-hackernews#2]
## PROBLEM: 
I'm following the book I bought via the Manning website (Testing Vue.js Applications), so far so great with the explanations but I can't pass the 2th chapter because of the following problem:
```
 FAIL  src/components/__tests__/Item.spec.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /home/stefano/lab/testingVueApps/vue-hackernews/src/components/__tests__/Item.spec.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import { shallowMount } from '@vue/test-utils';
                                                                                                    ^

    SyntaxError: Unexpected token {

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:403:17)

```
## Steps to reproduce:
Use a linux distro (I'm using ubuntu 18.04.2 right now, but happens the  same with version 16.04)
Install npm with 
Just follow the book instructions until chapter 2.

## Already tried with:
(I did follow step by step the instructions, carefully adding the transforms for jest in the package.json)
Checking that the package.json has the same versions as your repo.
 - result: same error
Using newer versions for jest related libraries
 - result: same error

For, the record, the first sanity check didn't pass until I followed the steps that were only explained later on (adding additional dependencies and mods to the package.json).

Thanks for taking a look as I can't continue with the book until I pass through this problem.

**Comment by StefanoA1** [comment]
@eddyerburgh

**Comment by eddyerburgh** [comment]
Hi Stefano, thank you for your detailed report. As you have found, since publication there have been problems with chapter 2, mainly because the packages aren't locked to a specific version. This was a mistake on my part—I should have asked readers to install specific versions to avoid future problems. I'm going to add an errata to fix this. I'll also add an errata to fix the syntax errors you have found. Sorry for the mistakes in the code.

**Comment by sevillaarvin** [comment]
https://manning-content.s3.amazonaws.com/download/b/921f670-42b2-4adb-ab94-70efb8bbf428/Yerburgh_Testingue.jsApplications_Err1.html

**Comment by eddyerburgh** [comment]
Thanks, again I'm sorry for these problems. I should have specified a version in the original text

**Comment by hoalongnatsu** [comment]
@sevillaarvin thanks you, i have same problem

**Comment by BThvedt** [comment]
Got stuck on this one for a bit, using the jest@23.6.0 by itself did not seem to work, but using Stephano's suggestion and replacing the chapter-2 package.json with chapter-3 branch's package.json file seemed to resolve it.

**Comment by torressam333** [comment]
Do you have the starting code for this book? I only see the completed project that isn't broken out into chapters when following along in the book.

**Comment by eddyerburgh** [comment]
@torressam333 you need to change branches. Each chapter X's starter code is at branch `chapter-X`

**Comment by torressam333** [comment]
@eddyerburgh yes thank you. IDK why but I was confusing branches with directories. My bad. Thank you, sir.
